app/javascript/components/upload/ExpressionFileForm.js

```diff
@@ -5,54 +5,67 @@ import Select from 'lib/InstrumentedSelect'
 import MTXBundledFilesForm from './MTXBundledFilesForm'
 import FileUploadControl, { FileTypeExtensions } from './FileUploadControl'
 import { TextFormField, SavingOverlay, SaveDeleteButtons } from './form-components'
+import { findBundleChildren, validateFile } from './upload-utils'
+
+const REQUIRED_FIELDS = [{ label: 'species', propertyName: 'taxon_id' },
```

Comment:
For now I understand that these are needed to control form logic for each individual form, but this just seems like an easy place for things to get out of sync with the Ruby models when we make changes.  A future enhancement could be an API endpoint that would serve this info.  We do have an existing one that might be good candidate/starting point (at `api/v1/schemas/study_files`) - we could pass in the `file_type` and get back the required fields.  It currently doesn't have any info right now about `expression_file_info`, so we'd need to add in that logic as well.

Reply:
yup